src/platform/utilities/environment/index.js

```diff
@@ -1,43 +1,85 @@
-const _Environments = {
-  preview: {
-    API_URL: 'https://api.va.gov',
-    BASE_URL: 'https://preview.va.gov',
-  },
-  vagovprod: {
-    API_URL: 'https://api.va.gov',
+/*
+ * Configuration for the current environment.
+ * @module platform/utilities/environment
+ */
+
+import ENVIRONMENTS from '../../../site/constants/environments';
+
+// eslint-disable-next-line no-undef
+const BUILDTYPE = __BUILDTYPE__;
+
+const {
+  WEB_PORT: RESERVED_E2E_PORT = 3333,
+  API_PORT: E2E_API_PORT = 3000,
+} = process.env;
+
+const ENVIRONMENT_CONFIGURATIONS = {
+  [ENVIRONMENTS.VAGOVPROD]: {
+    BUILDTYPE: ENVIRONMENTS.VAGOVPROD,
     BASE_URL: 'https://www.va.gov',
+    API_URL: 'https://api.va.gov',
   },
-  vagovstaging: {
-    API_URL: 'https://staging-api.va.gov',
+
+  [ENVIRONMENTS.VAGOVSTAGING]: {
+    BUILDTYPE: ENVIRONMENTS.VAGOVSTAGING,
     BASE_URL: 'https://staging.va.gov',
```

Comment:
We could derive the `BASE_URL` by importing `site/constants/hostnames` or even by passing `BUILD_OPTIONS.hostUrl` from Metalsmith into Webpack, but I preferred the explicitness of having hardcoded values on the Webpack side of things.